department-of-veterans-affairs/caseflow
# Package Document Type column APPEALS-40970

Resolves [APPEALS-40970](https://jira.devops.va.gov/browse/APPEALS-40970)

# Description
Please explain the changes you made here.
- Adds Package Document Type Column to all tabs on the Correspondence Cases table
- Adds Package Document Type Column to all tabs on the Your Correspondence table
- Implements column sorting on Package Document Type column
- Implements column filtering on Package Document Type column
- Implements search filtering within Package Document Type column

## Acceptance Criteria

## Testing Plan
1. Navigate to correspondence/queue using user Jolly_postman or AMBRISVACO.
2. Note that the Package document type column now exists on all tabs.
3. Toggle the sort and note that it sorts correctly.
4. Use the filter functionality for NOD or Non-NOD. Note that the filter applied and now shows only the selected type.
5. Use the filter functionality to add the other type. Note that both types now show.
6. Use the search bar to search for "-nod". note that the table reflected to only be Non-NOD.


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ![image](https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/8160e9b6-77a6-4fd4-acf4-e436796e534b)|<img width="1295" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/b1e53b62-dd1f-4542-989b-43d3c6d23c72">

# Backend
## Database Changes
*Only for Schema Changes*

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
